### PR TITLE
Create templates with member ToolchainCluster-related manifests

### PR DESCRIPTION
**Pull request by mfrancisc**
This PR adds only the manifests for the member toolchaincluster SA, ClusterRole and ClusterRoleBinding. The functionality for applying the manifests in the cluster will be implemented in follow up PRs.

For now there's a common functionality that parses those manifests from `embed.FS` filesystem.
https://github.com/codeready-toolchain/toolchain-common/pull/365

Those resources are now applied by the `add-cluster.sh` script, see: https://github.com/codeready-toolchain/toolchain-cicd/blob/34ea85fe21ae05c9bb1a0a2bbdc986b94c8d9e92/scripts/add-cluster.sh#L36

Jira: https://issues.redhat.com/browse/KSPACE-7

Related PRs:
common: https://github.com/codeready-toolchain/toolchain-common/pull/365
host-operator: https://github.com/codeready-toolchain/host-operator/pull/974

### CODE REVIEW

File: deploy/toolchaincluster/member-sa.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+rules:
+- apiGroups:
+  - authentication.k8s.io
+  resources:
+  - tokenreviews
+  verbs:
+  - create
+- apiGroups: [""]
+  resources: ["users", "groups"]
+  verbs: ["impersonate"]
```
Comment (alexeykazakov) [inline deploy/toolchaincluster/member-sa.yaml:33]: Or maybe just for consistency's sake so we use the same format in this template:
```suggestion
- apiGroups:
  - ""
  resources:
  - users
  - groups
  verbs:
  - impersonate
```

**Review by alexeykazakov — APPROVED** [review]
Looks good. Just one very minor comment.

File: deploy/toolchaincluster/member-sa.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  - "memberstatuses"
+  - "toolchainclusters"
+  - "useraccounts"
+  verbs:
+  - "*"
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: toolchaincluster-member-{{.Namespace}}-toolchaincluster
```
Comment (MatousJobanek) [inline deploy/toolchaincluster/member-sa.yaml:23]: I know that we have this crazy long names created by the script, but we could maybe make it shorter and clean the old ones manually. It shouldn't have any impact on the functionality of the controllers - it's just ClusterRole & ClusterRoleBinding, we don't change anything in the ServiceAccount.
```suggestion
  name: toolchaincluster-{{.Namespace}}
```
WDYT?

File: deploy/toolchaincluster/member-sa.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+rules:
+- apiGroups:
+  - toolchain.dev.openshift.com
+  resources:
+  - "idlers"
+  - "nstemplatesets"
+  - "memberoperatorconfigs"
+  - "memberstatuses"
+  - "toolchainclusters"
+  - "useraccounts"
```
Comment (MatousJobanek) [inline deploy/toolchaincluster/member-sa.yaml:16]: Let's do the same as we did for host, just use the wildcard:
```suggestion
  - "*"
```

File: deploy/toolchaincluster/member-sa.yaml (hunk 1 above)
Comment (mfrancisc) [reply deploy/toolchaincluster/member-sa.yaml:28]: thanks!

File: deploy/toolchaincluster/member-sa.yaml (hunk 2 above)
Comment (mfrancisc) [reply deploy/toolchaincluster/member-sa.yaml]: makes sense, I didn't want to change those names initially. 

Done in https://github.com/codeready-toolchain/member-operator/pull/537/commits/5b6bdb04ebcbfbf5b1c1648dc1b7c7e768def36b

Thanks!

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**